### PR TITLE
add new functions for v3 data

### PR DESCRIPTION
**Pull request by cshanes**
Hi, added new functions for the upcoming data release. 

These are safe to merge and release, the new download dataset endpoint is currently gated so only numerai employees can access it.

### CODE REVIEW

File: numerapi/numerapi.py [hunk 1]
```diff
@@ -1103,3 +1103,19 @@ def daily_submissions_performances(self, username: str) -> List[Dict]:
         performances = [p for p in performances
                         if any([p['correlation'], p['fnc'], p['mmc']])]
         return performances
+
+    def get_dataset_list(self):
```
Comment (uuazed) [inline numerapi/numerapi.py:1107]: I didn't know about listDatasets, quite useful!

Please document the method like all the other methods  in the repository.

File: numerapi/numerapi.py [hunk 2]
```diff
@@ -1103,3 +1103,19 @@ def daily_submissions_performances(self, username: str) -> List[Dict]:
...
+        query = """
+            query { listDatasets }
+        """
```
Comment (uuazed) [inline numerapi/numerapi.py:1110]: This doesn't need triple-quotes and should go on one line

File: numerapi/numerapi.py (hunk 1 above)
Comment (uuazed) [inline numerapi/numerapi.py:1107]: please add type hints

File: numerapi/numerapi.py [hunk 3]
```diff
@@ -1103,3 +1103,19 @@ def daily_submissions_performances(self, username: str) -> List[Dict]:
...
+        return self.raw_query(query)['data']['listDatasets']
+
+    def download_dataset(self, filename: str, dest_path: str):
```
Comment (uuazed) [inline numerapi/numerapi.py:1113]: again, missing doc string, missing type hints

Comment (uuazed) [reply numerapi/numerapi.py:1113]: Isn't this doing the same as  `Numerapi.download_latest_data`?  there is also `Numerapi.get_dataset_url`, which should be re-used

File: numerapi/numerapi.py [hunk 4]
```diff
@@ -1103,3 +1103,19 @@ def daily_submissions_performances(self, username: str) -> List[Dict]:
...
+        query = """
+        query ($filename: String!) {
+            dataset(filename: $filename)
+        }
+        """
+        dataset_url = self.raw_query(query, {'filename': filename}, authorization=True)['data']['dataset']
```
Comment (uuazed) [inline numerapi/numerapi.py:1119]: Does it really need authorization to get the dataset url?

**Review by uuazed — CHANGES_REQUESTED** [review]

File: numerapi/numerapi.py (hunk 4 above)
Comment (kumikoda) [reply numerapi/numerapi.py:1119]: yep - it does for now.

File: numerapi/numerapi.py (hunk 3 above)
Comment (kumikoda) [reply numerapi/numerapi.py:1113]: we should also get the round number and tournament number in here as an arg?

File: numerapi/numerapi.py (hunk 1 above)
Comment (kumikoda) [reply numerapi/numerapi.py:1107]: yep we just added this. it shows you all the individual files that you can download

File: numerapi/numerapi.py (hunk 2 above)
Comment (kumikoda) [reply numerapi/numerapi.py:1110]: lets get round number and tournament number in here as an arg?

File: numerapi/numerapi.py (hunk 3 above)
Comment (uuazed) [reply numerapi/numerapi.py:1113]: or, if the other two methods I mentioned are deprecated. Let's mark them as such

File: numerapi/numerapi.py (hunk 4 above)
Comment (uuazed) [reply numerapi/numerapi.py:1119]: ok, makes sense

**Comment by uuazed** [comment]
Closing this PR, the change landed in master already

**Pull request closed without merging**